### ISSUE TITLE
Dollar signs confuse the linter with links

### ISSUE THREAD

**Issue by mtlynch** [DavidAnson/markdownlint#1482]
I ran into a strange issue where dollar signs seem to confuse the linter into thinking a bare URL exists when it's actually a link:

```bash
$ cat example.md 
# Test

$4 is more than [$1](https://example.com).
```

```bash
$ markdownlint-cli2 example.md 
markdownlint-cli2 v0.17.2 (markdownlint v0.37.4)
Finding: example.md
Linting: 1 file(s)
Summary: 1 error(s)
example.md:3:22 MD034/no-bare-urls Bare URL used [Context: "https://example.com"]
```

**Comment by DavidAnson** [comment]
The rendering of this in the demo app indicates it is being purchased as a math expression and so the message is correct. You can backlash escape the first $ to avoid this.

https://dlaa.me/markdownlint/#%25m%23%20Test%0A%0A%244%20is%20more%20than%20%5B%241%5D(https%3A%2F%2Fexample.com).%0A

**Comment by mtlynch** [comment]
Ah, I see. Thanks for the clarification!
